From 3b86f427212c6ef9e14a9178ec398fe2e8a2da2e Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Sat, 5 Jan 2013 18:44:29 +0800
Subject: [PATCH 068/223] ENGR00238237-2 mx6sl: csi/v4l: Initialize the
 variable cam_fmt

This patch fixed the cam_fmt uninitialization issue.

Signed-off-by: LiGang <b41990@freescale.com>
Signed-off-by: Robby Cai <R63905@freescale.com>
(cherry picked from commit 48a48ea30c6e55e44c9eacaad316b5caa04a3dda)
---
 drivers/media/video/mxc/capture/csi_v4l2_capture.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/video/mxc/capture/csi_v4l2_capture.c b/drivers/media/video/mxc/capture/csi_v4l2_capture.c
index bcfbe09..5c9c69c 100644
--- a/drivers/media/video/mxc/capture/csi_v4l2_capture.c
+++ b/drivers/media/video/mxc/capture/csi_v4l2_capture.c
@@ -1740,6 +1740,7 @@ static int csi_v4l2_master_attach(struct v4l2_int_device *slave)
 	vidioc_int_dev_init(slave);
 	csi_enable_mclk(CSI_MCLK_I2C, false, false);
 	cam_fmt.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+	vidioc_int_g_fmt_cap(cam->sensor, &cam_fmt);
 
 	/* Used to detect TV in (type 1) vs. camera (type 0) */
 	cam->device_type = cam_fmt.fmt.pix.priv;
-- 
1.7.10.4

